### ISSUE TITLE
Infinite loop / memory exhaustion when acessing node in menu template

### ISSUE THREAD

**Issue by j6s** [neos/neos-development-collection#2532]
### Description

When accessing `{node}` in any way in a menu template (Site navigation, Breadcrumb, anything that inherits from `Neos.Neos:Menu`) then fusion will spin off into an infinite loop eventually resulting either in a maximum nesting level error (if you are using xdebug) or memory exhaustion (if not).

This is caused by `Neos.Neos:Menu` defining `node = ${this.node}`. This change was introduced when splitting menu and menu items prior to 4.3.

https://github.com/neos/neos-development-collection/blob/master/Neos.Neos/Resources/Private/Fusion/Prototypes/Menu.fusion#L14

This issue may be related to #2499 but the stack traces in that ticket are ever so slightly different from this - that's why this is a new issue with a defined incorrect behaviour.

### Steps to Reproduce

1. Add a menu to your page
```
mainMenu = Neos.Neos:Menu {
    templatePath = 'resource://Neos.Demo/Private/Templates/FusionObjects/MainMenu.html'
}
```

2. Try to access `{node}` in that template
```
<h1>{node.identifier}</h1>
```

#### Expected behavior

Blank page, memory exhaustion

#### Actual behavior

Node properties (in this example identifier) should be accessible

### Affected Versions

Neos: >= 4.3.0
- Change introducing the behaviour: https://github.com/neos/neos-development-collection/commit/a1cceaf6068b76c58fdffc44bc347441ba3e4434#diff-033bf88345978204633d4a60db6f410c
- Change removing the old behaviour: https://github.com/neos/neos-development-collection/commit/6f302f19d2d6dcf4293223d801d867d9fd1d208e#diff-033bf88345978204633d4a60db6f410cL5

**Comment by kitsunet** [comment]
Good catch, node = ${this.node} is definitely wrong...

**Comment by j6s** [comment]
Fixed with #2534
